lib/middleware/httpMethodAllowed.ts

```diff
@@ -0,0 +1,33 @@
+import { MiddlewareReturn } from "@lib/types";
+import { NextApiRequest, NextApiResponse } from "next";
+import { cors } from "./cors";
+
+/**
+ * Checks if the HTTP method of the request is part of the allowed list
+ * @param methods Array of acceptable HTTP methods
+ * @returns boolean, true if middleware blocked the request
+ */
+
+export const httpMethodAllowed = (methods: string[]) => {
+  return async (req: NextApiRequest, res: NextApiResponse): Promise<MiddlewareReturn> => {
+    try {
+      const requestBody = req.body && Object.keys(req.body).length > 0 ? req.body : undefined;
+      const method = requestBody?.method ? requestBody.method : req.method;
+      if (!methods.includes(method)) {
+        res.status(403).json({ error: "HTTP Method Forbidden" });
+        return { pass: false };
+      }
+
+      // Also check CORS by evaluating the origin of the request
+      // TODO: Modify origin to acceptable parameter
+      const blockedByCors = await cors({ origin: "*", methods: methods })(req, res);
```

Comment:
My assumption was that when the httpCheckMethod middleware is used it would also always be protected by CORS.  While we are already checking for the http method the inclusion of origin might something we want to include later on for requests originating from a browser.  It can be easily ripped out if we don't want to have any CORS checks for browserside calls.

Reply:
Alright no worries we can leave it in. Was just wondering if it would be better to leave it out so as to follow a single responsibility pattern for middlewares